### DISCUSSION TITLE
Welcome to AutoShutdownSchedule Discussions!

### DISCUSSION THREAD

**Discussion by tomasrudh** [tomasrudh/AutoShutdownSchedule#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Praveen-Pandeti-BY** [comment]
Hi @tomasrudh ,

Thank you very much for your solution, very useful. Just want to check with you, we have 1000+ VM's scattered across subscription so wondering have you tried with -parallel option to reduce the amount of time it take to run or any other recommendations .

Thanks,
Praveen

**Comment by tomasrudh** [comment]
Hello Praveen,

The script do start and stop machines in parallel, the option to the cmdlet Start-AzVM is called -NoWait.

With that many machines you might want to divide them in groups, in some way, and run multiple scripts. We have our machines split in different subscriptions, each running its own AutoShutdownSchedule. At most one script serves 137 machines, a run takes just under 5 minutes.

Regards,
Tomas

**Reply by Praveen-Pandeti-BY** [reply]
Thank you Tomas.

I will try with multiple scripts by subscription. Will update you on the result.

Regards,
Praveen
